plugins/adrv9002.c

```diff
@@ -965,7 +963,20 @@ static void adrv9002_check_channel_status(struct plugin_private *priv,
 	 * widget as the channel is the same for all widgets on this port...
 	 */
 	ret = iio_channel_attr_read_double(chan->w[0].chn, "rf_bandwidth", &dummy);
-	if (ret == -ENODEV) {
+
+	if(ret < 0 && ret != -ENODEV) {
+		printf("Warning: iio channel returned an error when reading it: %d. We assume the channel is enabled\n",
+		       ret);
+	}
+
+	return ret != -ENODEV;
+}
+
+static void adrv9002_check_channel_status(struct plugin_private *priv,
+					  struct adrv9002_common *chan,
+					  const char *gtk_str)
+{
+	if (adrv9002_channel_is_enabled(chan)) {
```

Comment:
You should separate the commits. One to introduce the new adrv9002_check_channel_status() with the needed changes in `adrv9002_check_channel_status()` and explaining we will have more places to call that new helper in a following patch. It makes the git history better and helps in reviewing (less changes per patch)